src/dataFlattener.ts

```diff
@@ -0,0 +1,66 @@
+// Needs typing (I originally wrote this in javascript)
+
+export const dataFlattener = (nestedData) => {
+  type flatData = {
+    code: string;
+    name: string;
+    type: string;
+    country: string;
+    subnational1: string | null;
+    subnational2: string | null;
+    fullHierarchyName: string;
+  };
+
+  type region = {
+    code: string;
+    name: string;
+    subnational1: string | undefined;
+    subnational2: string | undefined;
+  };
```

Comment:
**nit:** You should probably avoid creating types _inside_ a function, otherwise they'll be re-created every time the function is called which is unnecessary for our case. 

These can just live at the module level (i.e just in the file, not inside another function)

Reply:
Types are now outside of the function.